### PR TITLE
Basic primary flow circleCi setup

### PR DESCRIPTION
**Pull request by langsmith**
This pr adds a basic CircleCI config file for Firebase testing and other `primary` workflow steps. The config file _doesn't_ include Play Store publishing. That will be addressed in a separate pr. I'd like to merge this into `master` so that I can re-base other open prs with this pr's config file

This pr also sets up most of the needed changes for [the Gradle publishing plugin](https://github.com/Triple-T/gradle-play-publisher)

### CODE REVIEW

File: MapboxAndroidDemo/build.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.application'
 apply from: "$project.rootDir/gradle/script-git-version.gradle"
 apply plugin: 'com.google.firebase.firebase-perf'
+apply plugin: 'com.github.triplet.play'
```
Comment (tobrun) [inline MapboxAndroidDemo/build.gradle]: if release integration will not happen as part of this pr, please remove any parts of it

File: build.gradle [hunk 2]
```diff
@@ -1,34 +1,40 @@
...
+    dependencies {
+        classpath pluginDependencies.gradle
+        classpath pluginDependencies.firebase
+        classpath pluginDependencies.gradlePlayPublisher
+    }
+
+}
+
+plugins {
+    id 'com.palantir.circle.style' version '1.1.2'
```
Comment (tobrun) [inline build.gradle]: could you elaborate what above plugin is?

File: build.gradle [hunk 3]
```diff
@@ -1,34 +1,39 @@
...
-    maven { url "https://jitpack.io" }
-    maven { url "http://oss.sonatype.org/content/repositories/snapshots/" }
-    maven { url "https://maven.google.com" }
-  }
+    repositories {
+        jcenter()
+        google()
+        maven { url "https://jitpack.io" }
+        maven { url "http://oss.sonatype.org/content/repositories/snapshots/" }
+        maven { url "https://maven.google.com" }
```
Comment (Guardiola31337) [inline build.gradle]: If we're including `google()` this is not necessary anymore.

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          shell: /bin/bash -euo pipefail
+          command: |
+            echo "${GCLOUD_SERVICE_ACCOUNT_JSON}" > secret.json
+            gcloud auth activate-service-account --key-file secret.json --project mapbox-android-demo
+            rm secret.json
+      - run:
+          name: Export secrets for developer-config.xml
+          command: |
+            #!/bin/bash
+            echo "${MAPBOX_DEVELOPER_CONFIG}" > SharedCode/src/main/res/values/developer-config.xml
```
Comment (Guardiola31337) [inline circle.yml]: We could add `MAPBOX_DEVELOPER_CONFIG` within the [`secretKeysSetup` task](https://github.com/mapbox/mapbox-android-demo/blob/master/SharedCode/gradle-config.gradle) and remove this step.

**Pull request merged**